ZenPacks/zenoss/Microsoft/Windows/jobs.py

```diff
@@ -75,17 +76,19 @@ def getJobDescription(self, *args, **kwargs):
 
     def _run(self, **kwargs):
         org = self.dmd.Devices.getOrganizer('/Server/Microsoft')
-        self.log.info('Searching for and removing incompatible winrmservices from devices.')
-        devices_cleaned = 0
-        for device in org.getSubDevicesGen():
-            finished = False
-            devices_cleaned += 1
-            # This could take several passes
-            while not finished:
-                self.remove_winrmservices(device)
-                finished = False if device.componentSearch(meta_type='WinRMService') else True
-            if not devices_cleaned % 25:
-                self.log.info('Committing devices with unclean winrmservices removed.')
+        results = ICatalogTool(org).search(WinService, query=Eq('meta_type', 'WinRMService'))
```

Comment:
I don't think there's much performance benefit to using the catalog here instead of just iterating over all Windows devices and their os.winrmservices relationship. I think there's a chance that there will be objects in the os.winrmservices relationship that have a meta_type that isn't WinRMService that will be missed by this approach.

I think an approach like the following would be more likely to get rid of all device.os.winrmservices objects.

```
from Products.Zuul.interfaces import ICatalogTool
from ZenPacks.zenoss.Microsoft.Windows.BaseDevice import BaseDevice

device_results = ICatalogTool(dmd.Device).search(BaseDevice)
for device_result in device_results:
    # remove all device_result.getObject().os.winrmservices objects
```

Also, the `progresslog` module is already in the Windows ZenPack. So you might want to consider using it for this instead of your own modulo progress logging. See the ResetOSProcesses migrate script for an example.